Src/BootCamp.Chapter/CaesarCipher.cs

```diff
@@ -1,22 +1,243 @@
+        /// <summary>
+        /// Prints decrypted messages for each possible key available in input array.
+        /// </summary>
+        /// <param name="encryptedMessage">Message to be decrypted</param>
+        public static void PrintDecryptedVariants(string encryptedMessage)
         {
-            return message;
+            bool arguementIsValid = Utils.IsStringValid(encryptedMessage);
+            if (!arguementIsValid)
+            {
+                Console.WriteLine("Argument is not valid");
+                return;
+            }
+
+            var characterOccurrences = AnalyseFrequency(encryptedMessage);
+            var possibleKeys = FindPossibleKeys(characterOccurrences);
+
+            Console.WriteLine($"{possibleKeys.Length} possible key(s) found.{Environment.NewLine}");
+            Console.WriteLine($"Adjust the bellow settings for better results!");
+            Console.WriteLine($"KeyAccuracy: {KeyAccuracy}");
+            Console.WriteLine($"Top Character: {TopCharacter}");
+            Console.WriteLine($"BaseCharacter: {BaseCharacter}");
+            Console.WriteLine();
+
+            Console.WriteLine($"Decryption results: ");
+            foreach (var key in possibleKeys)
+            {
+                var crackedMessage = Decrypt(encryptedMessage, key);
+
+                Console.WriteLine($"Key = {key} - {crackedMessage}");
+                Console.WriteLine();
+            }
+        }
+
+        /// <summary>
+        /// Transfers the contents of two one dimension arrays to one bi-dimensional array.
+        /// </summary>
+        /// <param name="array1">First int one dimension input array.</param>
+        /// <param name="array2">Second int one dimension input array.</param>
+        /// <returns>A new Bi-dimensional int array created from array1 and array2.</returns>
+        private static int[][] Construct2dArray(int[] array1, int[] array2)
+        {
+            if (ArrayOps.IsNullOrEmpty(array1) || ArrayOps.IsNullOrEmpty(array2))
+            {
+                System.Console.WriteLine("Input arrays are not valid!");
+                return default;
+            }
+
+            int[][] newBiDiArr = new int[][] { array1, array2 };
+            for (int i = 0; i < array1.Length; i++)
+            {
+                newBiDiArr[0][i] = array1[i];
+            }
+            for (int i = 0; i < array2.Length; i++)
+            {
+                newBiDiArr[1][i] = array2[i];
+            }
+
+            return newBiDiArr;
+        }
+
+        /// <summary>
+        /// Transfers the elements from one of the arrays in the bi-dimensional array to one dimension array.
+        /// </summary>
+        /// <param name="inputArray">The array from witch the transfer</param>
+        /// <param name="arrayNumber"></param>
+        /// <returns>One dimension array based on argument 0 or 1</returns>
+        private static int[] Deconstruct2dArray(int[][] inputArray, int arrayNumber)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename method 'Deconstruct2dArray' to match pascal case naming rules, consider using 'Deconstruct2DArray'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012201)